x-pack/plugins/fleet/public/applications/fleet/sections/settings/components/download_source_flyout/confirm_update.tsx

```diff
@@ -15,52 +15,68 @@ import { getCountsForDownloadSource } from './services/get_count';
 }
 
 const ConfirmDescription: React.FunctionComponent<ConfirmDescriptionProps> = ({
   downloadSource,
   agentCount,
   agentPolicyCount,
-}) => (
-  <FormattedMessage
-    id="xpack.fleet.settings.updateDownloadSourceModal.confirmModalText"
-    data-test-subj="editDownloadSourcesConfirmModal.confirmModalText"
-    defaultMessage="This action will update {downloadSourceName} agent binary source. It will update {policies} and {agents}. This action can not be undone. Are you sure you wish to continue?"
-    values={{
-      downloadSourceName: <strong>{downloadSource.name}</strong>,
-      agents: (
-        <strong>
-          <FormattedMessage
-            id="xpack.fleet.settings.updateDownloadSourceModal.agentsCount"
-            defaultMessage="{agentCount, plural, one {# agent} other {# agents}}"
-            values={{
-              agentCount,
-            }}
-          />
-        </strong>
-      ),
-      policies: (
-        <strong>
-          <FormattedMessage
-            id="xpack.fleet.settings.updateDownloadSourceModal.agentPolicyCount"
-            defaultMessage="{agentPolicyCount, plural, one {# agent policy} other {# agent policies}}"
-            values={{
-              agentPolicyCount,
-            }}
-          />
-        </strong>
-      ),
-    }}
-  />
-);
+}) =>
+  agentCount !== undefined && agentPolicyCount !== undefined ? (
+    <FormattedMessage
+      id="xpack.fleet.settings.updateDownloadSourceModal.confirmModalText"
+      data-test-subj="editDownloadSourcesConfirmModal.confirmModalText"
+      defaultMessage="This action will update {downloadSourceName} agent binary source. It will update {policies} and {agents}. This action can not be undone. Are you sure you wish to continue?"
+      values={{
+        downloadSourceName: <strong>{downloadSource.name}</strong>,
+        agents: (
+          <strong>
+            <FormattedMessage
+              id="xpack.fleet.settings.updateDownloadSourceModal.agentsCount"
+              defaultMessage="{agentCount, plural, one {# agent} other {# agents}}"
+              values={{
+                agentCount,
+              }}
+            />
+          </strong>
+        ),
+        policies: (
+          <strong>
+            <FormattedMessage
+              id="xpack.fleet.settings.updateDownloadSourceModal.agentPolicyCount"
+              defaultMessage="{agentPolicyCount, plural, one {# agent policy} other {# agent policies}}"
+              values={{
+                agentPolicyCount,
+              }}
+            />
+          </strong>
+        ),
+      }}
+    />
+  ) : (
+    <FormattedMessage
+      id="xpack.fleet.settings.updateDownloadSourceModal.confirmModalTextWithoutCount"
+      data-test-subj="editDownloadSourcesConfirmModal.confirmModalText"
+      defaultMessage="This action will update {downloadSourceName} agent binary source. It will update related policies and agents. This action can not be undone. Are you sure you wish to continue?"
```

Comment:
I think that repeating the `will update` part twice sounds a bit strange here. Could we change to something like this instead?
`This action will update {downloadSourceName} agent binary source, its related policies and agents. This action can not be undone. Are you sure you wish to continue?`